strict-containers/src/Data/FingerTree/Strict.hs

```diff
@@ -73,9 +73,8 @@ infixl 5 |>
 -- wrap the original "Data.FingerTree" functions while forcing the provided
 -- value to WHNF.
 newtype StrictFingerTree v a = SFT {fromStrict :: FT.FingerTree v a}
-  deriving (Eq, Ord, Show)
-
-deriving newtype instance Foldable (StrictFingerTree v)
+  deriving (Eq, Ord, Show, Generic)
```

Comment:
Deriving `Generic` allows to break the abstraction barrier, in particular get an iso `StrictFingerTree ↔ FingerTree` (which allows thunks in the values to sneak back in). Are you intending to extensively use the `Generic` instance?